codalab/worker/worker.py

```diff
@@ -759,6 +775,29 @@ def bundle_service_reply(self, socket_id, err, message, data):
         else:
             self.bundle_service.reply(self.id, socket_id, message)
 
+    def start_stage_stats(self, uuid: str, stage: str) -> None:
+        """
+        Set the start time for a bundle in a certain stage.
+        """
+        self.runs[uuid].bundle_profile_stats[stage]['start'] = time.time()
+
+    def end_stage_stats(self, uuid: str, stage: str) -> None:
+        """
+        Set the end time for a bundle finishing a stage.
+        Set the elapsed time to the end time minus the start time.
+        """
+        self.runs[uuid].bundle_profile_stats[stage]['end'] = time.time()
+        self.runs[uuid].bundle_profile_stats[stage]['elapsed'] = (
+            self.runs[uuid].bundle_profile_stats[stage]['end']
+            - self.runs[uuid].bundle_profile_stats[stage]['start']
+        )
+
+    def init_stage_stats(self) -> Dict:
+        """
+        Returns a stage stats dict with default empty values for start, end, and elapsed.
+        """
+        return {'start': 0, 'end': 0, 'elapsed': None}
```

Comment:
Let's use None as a deafult
```suggestion
        return {'start': None, 'end': None, 'elapsed': None}
```